ophyd/controls/signal.py

```diff
@@ -423,20 +450,25 @@ def describe(self):
         dict
             Dictionary of name and formatted description string
         """
-        return {self.name: {'source': 'PV:{}'.format(self._read_pv.pvname),
-                            'dtype': 'number',
+        return {self.name: {'source': 'PV:%s' % self._read_pv.pvname,
+                            'dtype': self.dtype,
+                            'formatter': self.formatter,
                             'shape': []}}
 
     def read(self):
-        """Read the signal and format for data collection
+        """Read the signal and combine it with its timestamp.
+
+        This value is formatted according to the values in self.dtype and
+        self.format
 
         Returns
         -------
         dict
             Dictionary of value timestamp pairs
+            {'value': value, 'timestamp': timestamp}
         """
-
-        return {self.name: {'value': self.value,
+        value = self.describe()['dtype'](self.formatter % self.value)
```

Comment:
I am :-1: on doing this here, we should not be mutating the data at this layer.  We can use the `formatter` at the UI layer to make sure we don't display all the extra points.